tests/custom_ekus.rs

```diff
@@ -0,0 +1,65 @@
+static ALL_SIGALGS: &[&webpki::SignatureAlgorithm] = &[
+    &webpki::ECDSA_P256_SHA256,
+    &webpki::ECDSA_P256_SHA384,
+    &webpki::ECDSA_P384_SHA256,
+    &webpki::ECDSA_P384_SHA384,
+    &webpki::ED25519,
+    &webpki::RSA_PKCS1_2048_8192_SHA256,
+    &webpki::RSA_PKCS1_2048_8192_SHA384,
+    &webpki::RSA_PKCS1_2048_8192_SHA512,
+    &webpki::RSA_PKCS1_3072_8192_SHA384,
+];
+
+#[cfg(feature = "alloc")]
+fn check_cert(
+    ee: &[u8],
+    ca: &[u8],
+    eku: webpki::ExtendedKeyUsage,
+    result: Result<(), webpki::Error>,
+) {
+    let anchors = vec![webpki::TrustAnchor::try_from_cert_der(ca).unwrap()];
+
+    let time = webpki::Time::from_seconds_since_unix_epoch(0x1fed_f00d);
+    let cert = webpki::EndEntityCert::try_from(ee).unwrap();
+
+    assert_eq!(
+        cert.verify_is_valid_cert_with_eku(eku, ALL_SIGALGS, &anchors, &[], time),
```

Comment:
For the test certs you're using in `verify_custom_eku` I think you can drop `ALL_SIGALGS` and just specify `&[&webpki::RSA_PKCS1_2048_8192_SHA256]` here.